internal/transport/http2_client.go

```diff
@@ -1627,12 +1632,12 @@ func (t *http2Client) readServerPreface() error {
 // reader verifies the server preface and reads all subsequent data from
 // network connection.  If the server preface is not read successfully, an
 // error is pushed to errCh; otherwise errCh is closed with no error.
-func (t *http2Client) reader(errCh chan<- error) {
+func (t *http2Client) readerUtil(errCh chan<- error) error {
 	defer close(t.readerDone)
```

Comment:
Hmm, how about instead of another function and the awkward naming problem, do:

```go
func (t ...) reader(....) {
	var err error
	defer func() {
		close(t.readerDone)
		if err != nil {
			t.Close(err)
		}
	}()
	// ...
}
```